Dockerfile

```diff
@@ -9,4 +9,34 @@ RUN pip install --no-cache-dir \
     -r review/requirements.txt \
     ./review
 
+ENV LINTERS_DIRECTORY      /opt/linters
+
+ENV CHECKSTYLE_VERSION  8.44
+ENV CHECKSTYLE_DIRECTORY  ${LINTERS_DIRECTORY}/checkstyle
+
+ENV DETEKT_VERSION  1.14.2
+ENV DETEKT_DIRECTORY  ${LINTERS_DIRECTORY}/detekt
+
+ENV PMD_VERSION  6.36.0
+ENV PMD_DIRECTORY  ${LINTERS_DIRECTORY}/pmd
+
+RUN mkdir -p ${CHECKSTYLE_DIRECTORY} && mkdir -p ${DETEKT_DIRECTORY} && mkdir -p ${PMD_DIRECTORY}
+
+RUN apt -y update && apt -y upgrade
+
+# Install Curl and Unzip
+RUN apt -y install curl unzip
+
+# Install Detekt and Detekt-formatting
+RUN curl -sSLO https://github.com/detekt/detekt/releases/download/v${DETEKT_VERSION}/detekt-cli-${DETEKT_VERSION}.zip \
+    && unzip detekt-cli-${DETEKT_VERSION}.zip -d ${DETEKT_DIRECTORY} \
+    &&  curl -H "Accept: application/zip" https://repo.maven.apache.org/maven2/io/gitlab/arturbosch/detekt/detekt-formatting/${DETEKT_VERSION}/detekt-formatting-${DETEKT_VERSION}.jar -o ${DETEKT_DIRECTORY}/detekt-formatting-${DETEKT_VERSION}.jar
+
+# Install Checkstyle
+RUN curl -L https://github.com/checkstyle/checkstyle/releases/download/checkstyle-${CHECKSTYLE_VERSION}/checkstyle-${CHECKSTYLE_VERSION}-all.jar > ${CHECKSTYLE_DIRECTORY}/checkstyle-${CHECKSTYLE_VERSION}-all.jar
+
+# Install PMD
+RUN curl -sSLO https://github.com/pmd/pmd/releases/download/pmd_releases/${PMD_VERSION}/pmd-bin-${PMD_VERSION}.zip \
+    && unzip pmd-bin-${PMD_VERSION}.zip -d ${PMD_DIRECTORY}
+
```

Comment:
It would be better to move the linters installation before the hyperstyle code installation. This will allow you to use the Docker cache because the hyperstyle code changes more frequently than the linter versions.
See https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache

Or (maybe it even more better) move the linters installation to base docker image. In this case, you can continue to use the base image in GitHub actions.

Reply:
Should I do it in this PR? Or in another one?``

Reply:
@nbirillo, whatever works better for you.

Reply:
Let's do it in the next PR